### DISCUSSION TITLE
Does ESLint flat config with plugins supported?

### DISCUSSION THREAD

**Discussion by itsmnthn** [zed-industries/zed#9144] [Feature Requests]
I see there is support for ESLint and Flat config. I'm using `@antfu/eslint-config` to generate flat config for my Vue project. and have this zed settings but formatting / fixing is not respecting `eslint-plugin-vue`'s rules. Is this supported or still in wip?

```JSON
{
  "formatter": "language_server",
  "languages": {
    "JavaScript": {
      "code_actions_on_format": {
        "source.fixAll.eslint": true
      }
    },
    "TypeScript": {
      "code_actions_on_format": {
        "source.fixAll.eslint": true
      }
    }
  },
  "lsp": {
    "eslint": {
      "settings": {
        "codeActionOnSave": {
          "rules": [
            "import/order"
          ]
        }
      }
    }
  }
}
```

**Comment by mrnugget** [comment]
Can you give me an example config, example file, and tell me what it should format and what it shouldn't format?

**Comment by itsmnthn** [comment]
```javascript
// eslint.config.js

import antfu from '@antfu/eslint-config'

export default antfu(
  {
    // unocss: true,
    formatters: {
      css: false,
    },
  },
  {
    rules: {
      'vue/multi-word-component-names': 'off',
      'no-console': ['error', { allow: ['debug', 'error', 'timeEnd', 'time'] }],
      'node/prefer-global/process': 'off',
    },
  },
)
```

Formatted by Zed

```Vue
<script setup lang="ts">
// the formatting zed does
</script>

<template>
  <div class="w-full flex items-center">
    <div class="layout" :class="{
      'blocks': preferences.ui.style === 0,
      'no-blocks': preferences.ui.style === 1,
      'layout-default': preferences.ui.layout === 1,
    }">
      <!--  -->
    </div>
  </div>
</template>
```

Formatted by running manual command eslint [file.vue] --fix
```Vue
<script setup lang="ts">
// ...
</script>

<template>
  <div class="w-full flex items-center">
    <div
      class="layout" :class="{
        'blocks': preferences.ui.style === 0,
        'no-blocks': preferences.ui.style === 1,
        'layout-default': preferences.ui.layout === 1,
      }"
    >
      <!--  -->
    </div>
  </div>
</template>
```

A specific rule that is not being applied is this [vue/first-attribute-linebreak](https://eslint.vuejs.org/rules/first-attribute-linebreak.html)

**Comment by itsmnthn** [comment]
#9213 does this fix this problem? or it's nothing to do with this?

**Reply by mrnugget** [reply]
Depends on how it didn't work for you. What #9213 fixes is that no diagnostics would show up, none at all. If they did show up for you, but just not using your plugins, then it's a different thing.

**Reply by itsmnthn** [reply]
I've tested it the release doesn't solve this.

Do you think I should create a bug/feature issue for this issue?

**Reply by mrnugget** [reply]
Yeah, that'd be helpful. Create an issue with steps to reproduce, including configuration and example files.

**Reply by mrnugget** [reply]
@itsmnthn we now support flat configs: https://github.com/zed-industries/zed/issues/7271 It's on `main`.
